lnrpc/routerrpc/forward_interceptor.go

```diff
@@ -168,12 +171,11 @@ func (r *forwardInterceptor) resolveFromClient(
 }
 
 func (r *forwardInterceptor) handleFailMessage(failMessage []byte,
-	resolution *htlcswitch.FwdResolution) error {
+	encrypted bool, resolution *htlcswitch.FwdResolution) error {
 
-	// Verify that the size is equal to the fixed failure
-	// message size + hmac + two uint16 lengths. See BOLT
-	// #4.
-	if len(failMessage) !=
+	// Verify that for encrypted messages the size is equal to the fixed
+	// failure message size + hmac + two uint16 lengths. See BOLT #4.
+	if encrypted && len(failMessage) !=
```

Comment:
would it be reasonable to try and decode the message here if it's not encrypted? it would be the "equivalent" check

Reply:
What would be the purpose to try to decode? If you mean to avoid the new flag, I think that might be risky in case an encrypted failure message accidentally decodes.

Reply:
I didn't mean to avoid the flag! mostly a sanity check that the message is valid. if the interceptor is returning non-valid messages, it can be quite tricky to debug. I meant something like this:

```
if encrypted {
  // len check
} else {
 // decode check
}}
```

Reply:
Great suggestion, added